src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -4,109 +4,347 @@
+                {
+                    // Turn constants into initializers
+                    initializationValue = constantSchema.Value.Value != null ?
+                        BuilderHelpers.ParseConstant(constantSchema.Value.Value, propertyType) :
+                        Constant.NewInstanceOf(propertyType);
+                }
+                else if (IsStruct || parameterMap.ContainsKey(property.Declaration.Name.FirstCharToLowerCase()))
+                {
+                    // For structs all properties become required
+                    Constant? defaultParameterValue = null;
+                    if (property.SchemaProperty?.ClientDefaultValue is object defaultValueObject)
+                    {
+                        defaultParameterValue = BuilderHelpers.ParseConstant(defaultValueObject, propertyType);
+                        defaultInitializationValue = defaultParameterValue;
+                    }
+
+                    var inputType = TypeFactory.GetInputType(propertyType);
+                    if (defaultParameterValue != null && !TypeFactory.CanBeInitializedInline(property.ValueType, defaultParameterValue))
+                    {
+                        inputType = inputType.WithNullable(true);
+                        defaultParameterValue = Constant.Default(inputType);
+                    }
+
+                    var validate = property.SchemaProperty?.Nullable != true && !inputType.IsValueType ? ValidationType.AssertNotNull : ValidationType.None;
+                    var defaultCtorParameter = new Parameter(
+                        property.Declaration.Name.ToVariableName(),
+                        property.Description,
+                        inputType,
+                        defaultParameterValue,
+                        validate,
+                        null
+                    );
+
+                    initializationValue = defaultCtorParameter;
+                }
+                else
+                {
+                    initializationValue = null;// need to get discriminator value from here GetPropertyDefaultValue(property);
+
+                    if (initializationValue == null && TypeFactory.IsCollectionType(propertyType))
+                    {
+                        initializationValue = Constant.NewInstanceOf(TypeFactory.GetPropertyImplementationType(propertyType));
+                    }
+                }
+
+                if (initializationValue != null)
+                {
+                    defaultCtorInitializers.Add(new ObjectPropertyInitializer(property, initializationValue.Value, defaultInitializationValue));
+                }
+            }
+            return defaultCtorInitializers.ToArray();
+        }
+
+        protected override CSharpType? CreateInheritedType()
+        {
+            if (_inputModel.BaseModel is not null)
+                return new CSharpType(new ModelTypeProvider(_inputModel.BaseModel!, DefaultNamespace, null, _typeFactory));
```

Comment:
Should we get an existing model from the library here instead of creating a new model?
I believe it benefits if we could keep "for a specific model, we always use the same instance" if possible.
I think that this method is not invoked when the models are building, therefore it should be safe to find a model from the full list of models.